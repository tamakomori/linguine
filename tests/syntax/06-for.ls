func main() {
    // For-range loop
    for (i in 0..10) {
       print(i);
    }

    // For-value loop
    array = [];
    for (v in array) {
        print("v = " + v);
    }
    array = [1, 2, 3];
    for (v in array) {
        print("v = " + v);
    }

    // For-key-value loop
    dict = {};
    for (k, v in dict) {
        print("key = " + k + ", value = " + v);
    }
    dict = {aaa: "123", bbb: "456"};
    for (k, v in dict) {
        print("key = " + k + ", value = " + v);
    }
}
