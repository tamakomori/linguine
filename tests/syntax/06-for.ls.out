0
1
2
3
4
5
6
7
8
9
v = 1
v = 2
v = 3
key = bbb, value = 456
key = aaa, value = 123
